assets/js/map.js

```diff
@@ -0,0 +1,242 @@
+(function(exports) {
+
+  // our DOM container and status message
+  var map = d3.select("#team-map")
+        .classed("loading", true),
+      status = map.append("p")
+        .html("Loading the map...");
+
+  // names and geographic locations of airport codes
+  var locations = [
+    {code: "DCA", label: "Washington", location: [-77.037722, 38.852083]},
+    {code: "SFO", label: "San Francisco", location: [-122.374889, 37.618972]},
+    {code: "CHI", label: "Chicago", location: [-87.631667, 41.883611]},
+    {code: "DAY", label: "Dayton", location: [-84.219375, 39.902375]},
+    {code: "DEN", label: "Denver", location: [-104.673178, 39.861656]},
+    {code: "PHL", label: "Philadelphia", location: [-75.241139, 39.871944]},
+    {code: "NYC", label: "New York", location: [-74.005833, 40.714167]},
+    {code: "TUS", label: "Tuscon", location: [-110.941028, 32.116083]},
+    {code: "SEA", label: "Seattle", location: [-122.309306, 47.449]},
+    {code: "AUS", label: "Austin", location: [-97.669889, 30.194528]}
+  ];
+
+  // URLs to grab
+  var urls = {
+    team: "/api/team/api.json",
+    topology: "/assets/data/us-states.json"
+  };
+
+  // set up map size, projection, and <svg> container
+  var width = 1100,
+      height = 600,
+      proj = d3.geo.albersUsa()
+        .scale(1285)
+        .translate([width / 2, height / 2]),
+      svg = map.append("svg")
+        .attr("class", "map")
+        .attr("viewBox", [0, 0, width, height].join(" "));
+
+  // queue() loads the URLs in parallel, rather than
+  // using nested callbacks
+  queue()
+    .defer(d3.json, urls.team)
+    .defer(d3.json, urls.topology)
+    .await(allDone);
+
+  function allDone(error, team, topology) {
+    map.classed("loading", false);
+    if (error) return showError(error.statusText);
+
+    // kill the status message
+    status.remove();
+
+    // get a GeoJSON FeatureCollection from the topology
+    var states = topojson.feature(topology, topology.objects.states)
+      .features;
+    // draw the states background
+    renderMapBackground(states);
+
+    // group members by location
+    var members = d3.values(team)
+          .filter(function(d) { return d.location; }),
+        byLocation = d3.nest()
+          .key(function(d) { return d.location; })
+          .map(members);
+
+    // then assign a list of members to each location
+    locations.forEach(function(d) {
```

Comment:
Suggestion, it would be a little cleaner to make this function, so var updatedLocations = locations.map , etc.